app/generateSMTScriptProp.ts

```diff
@@ -0,0 +1,7 @@
+interface Smt { smt2: string; propositions: Set<string>; }
+
+export default function generateSMTScriptProp(userSmt: Smt, sysSmt: Smt): string {
+  const declarationsArray = Array.from(new Set([...userSmt.propositions, ...sysSmt.propositions])).map((d) => `(declare-const ${d} Bool)\n`);
```

Comment:
This line merges the sets into one (with no duplicates), then maps those into the kinds of strings we want.